pkg/shp/cmd/buildrun/create.go

```diff
@@ -73,10 +75,25 @@ func (c *CreateCommand) Run(params *params.Params, ioStreams *genericclioptions.
 	br := &buildv1alpha1.BuildRun{Spec: *c.buildRunSpec}
 	flags.SanitizeBuildRunSpec(&br.Spec)
 
+	var build = buildv1alpha1.Build{}
+	if err := resource.GetBuildResource(params).Get(c.cmd.Context(), c.buildRunSpec.BuildRef.Name, &build); err != nil {
+		return fmt.Errorf("failed to get referenced build %q: %w", c.buildRunSpec.BuildRef.Name, err)
+	}
+
+	// Local source code mode:
+	// Make sure to bundle the configured local source directory into an image
+	// bundle and push it to the provided target registry.
+	if build.Spec.Source.Container.Image != "" {
```

Comment:
I do not recall who, but someone told me about this and if I recall correctly the idea is that in Go (for some language implementation specific reasons) the `if x == ""` check is preferred check, because (and here my memory gets a bit fuzzy) the `""` is some kind of special `nil` for `string` or something like that. If somebody knows the nitty gritty details, I would be really interested in that. I kind of like those details.

Reply:
For what it is worth, from a purely aesthetic point of view, I like the `len(s) > 0` when reading code.